Makefile

```diff
@@ -1,14 +1,17 @@
 GOPATH ?= $(shell go env GOPATH)
+DLEQ_LIB=dleq/cgo-dleq/lib/libdleq.so
 
 .PHONY: all
 all: install
 
-.PHONY: init
-init:
+$(DLEQ_LIB):
 	./scripts/install-rust.sh
 	git submodule update --init --recursive
 	cd dleq/cgo-dleq && make build
 
+.PHONY: init
+init: $(DLEQ_LIB)
```

Comment:
I made the `init` target depend on the existence of `libdleq.so`, then the target commands previously tied to `init` will only be executed if the shared library does not exist.  That prevents those commands from being run every single time you execute commands like `make lint`. To force the rebuild/submodule-update, you can run `make clean init`.

Reply:
much nicer, thanks!